misc/generic_editor.py

```diff
@@ -236,30 +208,57 @@ def word_prev(m):
 
 ctx.keymap(
     {
-        # META
-        "sage": Key("cmd-s"),
-        "dizzle": Key("cmd-z"),
-        "rizzle": Key("cmd-shift-z"),
+        # meta
+        "(save it | sage)": Key("cmd-s"),
+        "(undo it | dizzle)": Key("cmd-z"),
+        "(redo it | rizzle)": Key("cmd-shift-z"),
+        # clipboard
+        "([clip] cut | snatch)": Key("cmd-x"),
+        "([clip] copy | stoosh)": Key("cmd-c"),
+        "([clip] paste | spark)": Key("cmd-v"),
         # MOTIONS
-        "spring" + optional_numerals: jump_to_eol_and(jump_to_beginning_of_text),
-        "dear" + optional_numerals: jump_to_eol_and(lambda: None),
-        "smear" + optional_numerals: jump_to_eol_and(jump_to_nearly_end_of_line),
-        # 'sprinkoon' + numerals: jump_to_eol_and(lambda: press('enter')),
-        "shockey": Key("ctrl-a cmd-left enter up"),
-        "shockoon": Key("cmd-right enter"),
-        "jolt": Key("ctrl-a cmd-left shift-down cmd-c down cmd-v"),  # duplicate line
-        # DELETING
-        "snipple": Key("shift-cmd-left delete"),
-        "snipper": Key("shift-cmd-right delete"),
-        "shackle": Key("cmd-right shift-cmd-left"),
-        "snipline" + optional_numerals: jump_to_bol_and(snipline),
-        # SELECTING
-        "sprinkle" + optional_numerals: select_line,
-        "crew <dgndictation>": select_text_to_right_of_cursor,
-        "trail <dgndictation>": select_text_to_left_of_cursor,
-        "shift home": Key("shift-home"),
-        "wordneck" + optional_numerals: word_neck,
-        "wordprev" + optional_numerals: word_prev,
-        "word this": [Key("alt-right"), Key("shift-alt-left")],
+        "(go word left | fame | peg)": Key("alt-left"),
+        "(go word right | fish | fran)": Key("alt-right"),
+        "(go after line end | derek)": Key("cmd-right space"),
+        "(go line start | spring | lefty) {generic_editor.n}*": select_line_and_press(
```

Comment:
as implemented they dont if you dont give a numeral

Reply:
It's sort of a style thing, but this method makes it so you must pause after saying lefty if you want to say any numbers. Others have also done similar things for go left so that go left 5 types left 5 times. It drove me crazy though, it kept eating my numbers. Maybe it should be a global style option of some kind?